ffi/python/citi/record.py

```diff
@@ -0,0 +1,296 @@
+'''Record implemented through a foreign function interface
```

Comment:
PR #19 adds support for error handling through the ffi that can be take advantage of on the python side.